### PR TITLE
PLATUI-1941: Added timeout-dialog synchronise tabs

### PR DESCRIPTION
**Pull request by JoPintoPaul**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -767,6 +767,16 @@ The instructions below assume you have set up play-frontend-hmrc as indicated ab
    a parameter. For example if using `hmrcLayout`, pass `Some(hmrcTimeoutDialogHelper(signOutUrl = signOutUrl))` in the 
    `additionalHeadBlock` parameter.
 
+#### Synchronise session between tabs ####
+
+Additionally, services can choose to opt-in to behaviour to synchronise session extension between different HMRC tabs
+(using the `BroadcastChannel` API in browsers). In practical terms, this means that if a user sees a timeout dialog in
+an active tab, and clicks to extend their session, the session will also be extended in any background tabs.
```
Comment (oscarduignan) [inline README.md:774]: What about
```suggestion
an active tab, and clicks to extend their session, then the timeout dialogs that have also opted into this behaviour in any background tabs will also be restart the countdowns until they display their timeout warning.
```

File: src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/HmrcTimeoutDialogHelper.scala.html [hunk 2]
```diff
@@ -30,7 +30,8 @@
   message: Option[String] = None,
   messageSuffix: Option[String] = None,
   keepAliveButtonText: Option[String] = None,
-  signOutButtonText: Option[String] = None
+  signOutButtonText: Option[String] = None,
+  synchroniseTabs: Option[Boolean] = None
```
Comment (oscarduignan) [inline src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/HmrcTimeoutDialogHelper.scala.html:34]: is this possible?

```suggestion
  synchroniseTabs: Boolean = timeoutDialogConfig.enableSynchroniseTabs
```

Comment (oscarduignan) [reply src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/HmrcTimeoutDialogHelper.scala.html:34]: I guess it probably isn't, just spotted that other optional stuff is using option wrapped values!

Comment (JoPintoPaul) [reply src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/HmrcTimeoutDialogHelper.scala.html:34]: @oscarduignan Yes, I was thinking that `Option[Boolean]` captures that this doesn't have to be set by the user, i.e. is optional. Having said, I have very mixed feelings about `Option[Boolean]` under any circumstances, so more than happy to revisit this.

Comment (oscarduignan) [reply src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/HmrcTimeoutDialogHelper.scala.html:34]: I'm easy either way, the more I squint the more it seems like not much of a muchness

File: README.md [hunk 3]
```diff
@@ -767,6 +767,16 @@ The instructions below assume you have set up play-frontend-hmrc as indicated ab
    a parameter. For example if using `hmrcLayout`, pass `Some(hmrcTimeoutDialogHelper(signOutUrl = signOutUrl))` in the 
    `additionalHeadBlock` parameter.
 
+#### Synchronise session between tabs ####
+
+Additionally, services can choose to opt-in to behaviour to synchronise session extension between different HMRC tabs
+(using the `BroadcastChannel` API in browsers). In practical terms, this means that if a user sees a timeout dialog in
+an active tab, and clicks to extend their session, then the timeout dialogs that have also opted into this behaviour in any background tabs will also be restart the countdowns until they display their timeout warning.
```
Comment (oscarduignan) [inline README.md:774]:
```suggestion
an active tab, and clicks to extend their session, then the timeout dialogs that have also opted into this behaviour in any background tabs will also restart the countdowns until they display their timeout warning.
```

Sorry @JoPintoPaul I left a rogue word in there by mistake!

**Pull request merged**